File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>
#include <algorithm>
#include <iostream>
#include <queue>
using namespace std;

vector<int> solution(vector<int> arr){
   vector<int> answer;
    queue<int> q;

    for(int i = 0; i < arr.size(); i++){
        
        // 큐가 비어있을 때 넣음
        if(q.empty()){
            q.push(arr[i]);
        } else {
            if(arr[i] != q.back()){
                q.push(arr[i]);
            } else continue;
        }
    }
    
    while(!q.empty()){
        int num = q.front();
        q.pop();
        
        answer.push_back(num);
    }
    
    return answer;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public ArrayList <Integer> solution(int []arr) {
        ArrayList<Integer> answer = new ArrayList<>();
        
        // 첫번째 원소는 무조건 추가
        if(arr.length > 0){
            answer.add(arr[0]);
        }
        
        // 뒤에 수와 같지 않을 때만 answer에 추가
        for(int i = 1; i < arr.length; i++){
    
            if(arr[i] != arr[i-1]){
                answer.add(arr[i]);
            }
        }
        return answer;
    }
}